azure-iot-device/azure/iot/device/iothub/auth/base_renewable_token_authentication_provider.py

```diff
@@ -160,9 +171,12 @@ def _schedule_token_update(self, seconds_until_update):
             seconds_until_update,
         )
 
+        self_weakref = weakref.ref(self)
```

Comment:
>self_weakref = weakref.ref(self) [](start = 8, length = 32)

Could you provide an explanatory comment outlining the scenario in which this is relevant? Does the Auth Provider ever get deleted? My understanding was that it hangs around indefinitely in the transport stage that handles authentication.

Reply:
The auth provider should get cleaned up when the pipeline gets cleaned up, which happens when the client gets cleaned up.  Usually this is when you do a clean shutdown, but it can also happen if the user deletes their client object and creates a new one.

Because of various circular references (like callbacks, but also next and previous pointers), if the auth provider doesn't get collected, then the entire pipeline doesn't get collected.

I'm also adding a comment  to the code in a new revision.